plugins/module_utils/system_access_users_utils.py

```diff
@@ -780,17 +832,25 @@ def add_or_update(self, user: User) -> None:
         )
         next_uid: Element = self.get("uid")
 
-        # since the current password of an user cannot not be compared with the new one,
-        # we're setting the password anyways
-        self.set_user_password(user)
-
         if existing_user:
+
+            if not password_verify(
+                existing_user_password=existing_user.password, password=user.password
```

Comment:
I think it would make sense to add an attribute to the User class eg. a User.password_hash attribute. The tracking of when a password is hashed in an object instance and when not makes the code more complex to understand and introduces semantic inconsistencies into the code.